### ISSUE TITLE
Fix doc comment line splits for the first paragraph or report issue with tooling

### ISSUE THREAD

**Issue by chanced** [chanced/jsonptr#58]
Some doc comments have their first paragraph/line split between two lines. This is causing issues with rust analyzer on at least one editor:

<img width="645" alt="346714924-3aad60db-28ec-4761-9225-7265bf5123b3" src="https://github.com/chanced/jsonptr/assets/1073892/e978942d-6ea9-4a5a-b73d-d4fb5e2015b5">
<img width="970" alt="346715427-2a6afa74-d712-4057-ab36-7f6cb44110df" src="https://github.com/chanced/jsonptr/assets/1073892/6891d14e-96f8-4165-ae30-d6d988eb16d3">

Either need to report the issue, make the first paragraph/line a single line, or both.

**Comment by asmello** [comment]
I've clarified it with Jon, and as it turns out his remark about the first line having special treatment was about the first line under markdown semantics - delimited by `\n\n` - so I _think_ this is actually ok. It does seem unintended to have the newlines reproduced raw in the hover docs, but I tested with vscode and I can't reproduce, so it seems that `rust-analyzer` sends the raw markdown and it's up to the editor to render how they choose. So I guess this is on Helix to fix.

Sorry about the false alarm! Will close out, hope you haven't wasted energy changing this yet.

**Comment by chanced** [comment]
I dont think this was a false alarm, but rather that some tooling in the ecosystem isn't properly parsing markdown, or at least not rendering it properly.

The screenshots were from neovim right? I wrecked my config for neovim and just haven't had the time to go back and get it fixed. Kinda got sucked into the easiness of vscode with vim bindings and just got lazy, i guess.

**Comment by asmello** [comment]
> The screenshots were from neovim right? I wrecked my config for neovim and just haven't had the time to go back and rectify that. I kinda got sucked into the easiness of vscode with vim bindings and just got lazy, I guess.

Actually I use [Helix](https://helix-editor.com/)! If you're used to Neovim, Helix is probably not _quite_ mature enough yet for a migration, but since I never used Neovim in the first place I figured I may as well skip straight into the next thing.

**Comment by chanced** [comment]
Ah, helix is quite nice. I actually prefer the movements in it, the only reason I haven't switched is the lack of a plugin system. Well, that or a file tree (not sure why I'm so dependent on one).
